### ISSUE TITLE
Request to toggle diff fold using Enter

### ISSUE THREAD

**Issue by butterflyfish** [jreybert/vimagit#71]
currently, 'zo' can open diff fold while 'zc' can close diff fold.  I want to use press Enter to toggle diff fold.

my vimrc have snippet:

> nnoremap <expr> <CR> foldlevel(line('.'))  ? "za" : "<CR>"

**Comment by jreybert** [comment]
I suspect that the command `foldlevel(line('.'))` will only be evaluated once, at startup.

**Comment by dfishburn** [comment]
I use the following:

```
" Tip #108: Toggle a fold with a single keystroke {{{
" Toggle fold state between closed and opened.
" Based on Vim tip:
" http://www.vim.org/tips/tip.php?tip_id=108
" http://vim.wikia.com/wiki/VimTip108
" http://vim.wikia.com/wiki/Folding
" Explanation:
" - 'normal! za' toggles folds.
" - 'silent!' allows to avoid error message when current line doesn't belong
"   to fold.
" - (foldlevel('.')?'':'l') adds forward moving only if current line doesn't
"   belong to fold.
" nnoremap  <silent> <space> :exe 'silent! normal! za'.(foldlevel('.')?'':'l')<cr>
nnoremap  <silent> <space> :exe 'silent! normal! '.((foldclosed('.')>0)? "zMzxzt" : 'zc')<cr>
```

zM - resets the fold
zx - opens the fold
zt - moves the first line to the top of the screen

**Comment by butterflyfish** [comment]
@dfishburn , thx, but I want to use <enter>. I set <leader> to space key.
I hope vimagit let CR have multiple function.
- <CR>   if cursor on filename header line, unhide diffs for this file
- <CR>   if cursor on  diff hunks, toggle diff fold

As result, it can improve user experience. One key Enter is quick than zo/zc.

**Comment by mtglsk** [comment]
@dfishburn How does this mapping differ from `za`?

**Comment by jreybert** [comment]
There would be a solution for that request, but i can't decide if if is more natural to use `<cr>` to open/close vim folding or to navigate in normal mode.

**Comment by jreybert** [comment]
@butterflyfish could you try branch `dev/issue_71` please, and tell me if it is what you expect?

**Comment by jreybert** [comment]
Merged in master and present in release 1.7.2
